Feature/soundfx

WillTarte opened the pull request:
Added sound effects:
- Each gun has shooting & reloading (if applicable) (also includes throwing grenades)
- Explosions have sound effects
- Opening chests, and picking up ammo/health has sound effects
- Player has sound effects for: running, jumping, rolling, climbing, taking damage, switching weapons
- game over sound effect

WillTarte commented:
I'm pretty sure whoever reviews this will get errors or debug messages in the editor due to some sound files getting messed up by Git LFS. Just let me know it's probably an easy fix.

ClaudiaLapalme reviewed (changes requested):
There are a few changes unrelated to sound effects here that I don't quite understand or think should be in this PR

WillTarte commented on Assets/Scripts/EnemySystem/ScriptableObjects/EnemyAttackStrategies/GunnerStrategy.cs:
```diff
@@ -21,8 +21,7 @@ public class GunnerStrategy : EnemyAttackStrategy
         {
-            if ((Math.Abs(emu.transform.position.x) - Math.Abs(player.transform.position.x) < shootDistanceRangeMax)
-                && (Math.Abs(emu.transform.position.y) - Math.Abs(player.transform.position.y) < shootHeightRangeMax))
+            if (Vector2.Distance(player.transform.position, emu.transform.position) < shootDistanceRangeMax)
```
I decided to change this file because it was causing an issue with the sound effects where the gunner emu would shoot (you could hear it) but the enemy was super far away

ClaudiaLapalme replied:
oooooo i see, that's fine, then!

WillTarte commented on Assets/Scripts/Menus/MainMenu.cs:
```diff
@@ -1,10 +1,16 @@
     {
+        private void Awake()
+        {
+            PlayerPrefs.SetInt("volume", 7);
```
So that the player can change the volume for the whole game and not just the main menu

WillTarte commented on Assets/Scripts/Menus/OptionsMenu.cs:
```diff
@@ -9,7 +9,8 @@ public class OptionsMenu : MonoBehaviour
     
         public void SetVolume(int vol)
         {
+            PlayerPrefs.SetInt("volume", vol);
```
for volume mixer that's the case. But changing the volume for a AudioSource (which Is what I used) the volume is on a scale from 0 to 1 which can easily map with out scale of 0 to 10 (whole numbers)

ClaudiaLapalme replied:
Sounds good!

ClaudiaLapalme replied:
Did you remove the sound mixer, then?

WillTarte replied:
I didn't touch it cuz I don,t want to break anything on the Main Menu (it only exists in the main menu scene)

WillTarte commented on Assets/Scripts/Player/InputHandler.cs:
```diff
@@ -8,7 +8,7 @@ public class InputHandler
         private readonly Command _jumpCmd = new JumpCommand();
         private readonly Command _shootCmd = new ShootCommand();
         private readonly Command _reloadCmd = new ReloadCommand();
-        private readonly Command _pickupCommand = new InteractCommand();
```
Yeah I realized that it should be renamed to clear confusion

ClaudiaLapalme commented:
1. There should be a sound when the emu dies
2. There should be a sound when the player dies
3. The blood splatter when the player dies is no longer occurring for some reason???
4. The gun shooting sound is kind of wack (not a big deal)
5. Is there supposed to be a sound when chests open? I don't hear anything when I open chests or pick up its contents (there should be a sound for that as well imo)

WillTarte commented:
> 1. There should be a sound when the emu dies
> 2. There should be a sound when the player dies
> 3. The blood splatter when the player dies is no longer occurring for some reason???
> 4. The gun shooting sound is kind of wack (not a big deal)
> 5. Is there supposed to be a sound when chests open? I don't hear anything when I open chests or pick up its contents (there should be a sound for that as well imo)

2. There is already a sound (works for me)
3. I didn't touch it so idk what's going on
5. There's supposed to be a sound (works for me)

ClaudiaLapalme commented:
2. idk I still don't hear it, but there's a huge delay between dying and the game going to the death scene. Maybe play the sound where the blood splatter occurs to avoid adding a double delay (there's already a delay for the blood particles)
3. The blood splatter only occurs when the player gets killed by an emu, but not by other stuff. I'll log an issue for that
4. I meant ugly for the auto gun (still works)
6. I think there should be a sound when the player gets hurt by a spike
7. I've noticed that some sounds are way louder than others

WillTarte commented on Assets/Scripts/Player/SoundController.cs:
```diff
@@ -0,0 +1,151 @@
+        {
+            switch (newState)
+            {
+                case RunState _:
```
the default is that it the switch statement doesn't do anything

WillTarte commented:
@ClaudiaLapalme 
2. I didn't play around with delays. The sound is triggered alongside where the blood is instantiated. It works for me
3. oki dokes
5. There should be a sound already whenever the player takes damage
I added an emu death sound as well

Aleoris commented:
@ClaudiaLapalme  can you please confirm it is the same for you 
1. gun, reloading, and grenade works for me
2.  explosion works for me
3. chest, ammo/ health has no sound effects
4. no player sounds effects
5. no game over sound (actually, I don't have sound in the main menu too)

ClaudiaLapalme commented:
@Aleoris yep that's the same on my side. @WillTarte could you look into it before we resort to just taking the sounds from google drive? this is odd and should be fine considering that half of the sounds are there.

By player sound effects, you mean when they take damage/die, right?

Aleoris reviewed (changes requested):
1. When the player dies, the game waits for the dead sound to finish before changing the scene. Can you add the sound to the game over screen instead? so we don't stay stuck in the game 
2. There is no sound when picking up the ammo from the chest
3. The walking sound is really annoying, can you make it quieter or just change it xD
4. there is a bug when you kill an emu and keep shooting where the emu was, the sound and blood never stops (need to use assault riffle)

ClaudiaLapalme commented:
1. You should not destroy the player when you kill him, it deletes the camera and fudges shit up
2. I agree with Alessandro and the delay before transitioning to the death menu is like 1-2 seconds too long
3. I agree with Alessandro that there should be a sound for picking up shit from a chest
4. The walking sounds like knocking. I think we should change the sound and play it half as frequently (it plays twice per step atm)

WillTarte commented:
@ClaudiaLapalme @Aleoris I addressed your comments.
Regarding the chest pickups, there are already supposed to be sounds (it's the first thing I added sounds for). I also updated some of the sounds, so make sure you grab the folder on google drive

ClaudiaLapalme commented:
1. I would put the main menu song in the levels as well
 2. The footsteps sound like gun shots. I think that a different sound (and a lot quieter) would be a lot better

great job on the rest!!!! (I am dumb and commented on the wrong PR)

ClaudiaLapalme commented:
The shooting sounds and damage sounds are a lot louder than the other sounds. 

Other than that, this is looking pretty good!

ClaudiaLapalme reviewed (approved):
woooooooooooooo

Aleoris reviewed (approved):
The sound in the menu main is very loud, but it can be fixed later. Other than that... Perfection
